src/sqlfluff/core/templaters/base.py

```diff
@@ -168,18 +169,15 @@ def get_line_pos_of_char_pos(
             line_number, line_position
 
         """
-        nl_idx = -1
-
         if source:
             ref_str = self._source_newlines
         else:
             ref_str = self._templated_newlines
 
-        while nl_idx + 1 < len(ref_str) and ref_str[nl_idx + 1] < char_pos:
-            nl_idx += 1
+        nl_idx = bisect_right(ref_str, char_pos)
 
-        if nl_idx >= 0:
-            return nl_idx + 2, char_pos - ref_str[nl_idx]
+        if nl_idx > 0:
+            return nl_idx + 1, char_pos - ref_str[nl_idx - 1]
```

Comment:
The first change seems fine, but this change is causing some of the tests to fail so they musn't be equivalent.

Reply:
interesting, the lexer/parser tests passed (at least locally).. looks like this is failing in the linter. I'll take a peek Friday or Monday.

Reply:
Here's what I run locally:

```
tox -e generate-fixture-yml,cov-init,py38,cov-report,linting
```

Quite slow, but covers everything the CI does. Or just update branch and let GitHub do the hard work 😁 - which basically does the above, plus more versions of Python, plus dbt tests and also some doc linting.

Reply:
@tunetheweb can you kick the CI workflow? tests are passing locally now

Reply:
Running now!

Reply:
All tests passing. Late here now so will give it a once over over the weekend unless someone else gets in there first.

Not seeing any real noticeable improvement on the CI run time. Was hoping you were being modest about the improvements! 😁 Still every little helps!

Reply:
it's really query dependent, some of the larger ones we have saw a 10-20% improvement. it shouldn't make anything worse though. most queries spend so much time in the matcher lookahead logic that other minor perf bumps barely register.

Reply:
Sounds great!

Running the automated test suite is often a poor indicator of performance because tests involve lots of tiny examples, lots of setup and teardown, etc.